pyfar/dsp/filter/audiofilter.py

```diff
@@ -4,6 +4,134 @@
+        Filter characteristic coefficients ``bi`` and ``ai``.
+
+        -   For 1st order allpass provide ai-coefficient as single value.\n
+            The default is ``ai = 0.6436``.
+
+        -   For 2nd order allpass provide coefficients as list ``[bi, ai]``.\n
+            The default is ``bi = 1.6278``, ``ai = 0.8832``.
+
+        Defaults are chosen according to Tietze et al. (Fig. 12.66)
+        for maximum flat group delay.
+    sampling_rate : None, number
+        The sampling rate in Hz. Only required if signal is ``None``. The
+        default is ``None``.
+    Returns
+    -------
+    signal : Signal
+        The filtered signal. Only returned if ``sampling_rate = None``.
+    filter : FilterIIR
+        Filter object. Only returned if ``signal = None``.
+    References
+    ----------
+    .. [#] Tietze, U., Schenk, C. & Gamm, E. (2019). Halbleiter-
+        Schaltungstechnik (16th ed.). Springer Vieweg
+    .. [#] https://github.com/spatialaudio/digital-signal-processing-lecture/\
+blob/master/filter_design/audiofilter.py
+
+    Examples
+    -----
+    First and second order allpass filter with ``fc = 1000`` Hz.
+
+    .. plot::
+
+        import pyfar as pf
+        import matplotlib.pyplot as plt
+        import numpy as np
+
+        # impulse to be filtered
+        impulse = pf.signals.impulse(256)
+
+        orders = [1, 2]
+        labels = ['First order', 'Second order']
+
+        fig, (ax1, ax2) = plt.subplots(2,1, layout='constrained')
+
+        for (order, label) in zip(orders, labels):
+            # create and apply allpass filter
+            sig_filt = pf.dsp.filter.allpass(impulse, 1000, order)
+            pf.plot.group_delay(sig_filt, unit='samples', label=label, ax=ax1)
+            pf.plot.phase(sig_filt, label=label, ax=ax2, unwrap = True)
+
+        ax1.set_title('1. and 2. order allpass filter with fc = 1000 Hz')
+        ax2.legend()
+    """
+
+    # check input
+    if (signal is None and sampling_rate is None) \
+            or (signal is not None and sampling_rate is not None):
+        raise ValueError('Either signal or sampling_rate must be none.')
+
+    # check if coefficients match filter order
+    if coefficients is not None and (
+            (order == 1 and np.isscalar(coefficients) is False)
+            or (order == 2 and (isinstance(coefficients, (list, np.ndarray))
+                                is False or len(coefficients) != 2))):
+        print(type(coefficients), order)
+        raise ValueError('Coefficients must match the allpass order')
+
+    # sampling frequency in Hz
+    fs = signal.sampling_rate if sampling_rate is None else sampling_rate
+
+    if order == 1:
+        if coefficients is None:
+            coefficients = 0.6436
+        # get filter coefficients for first order allpass
+        _, _, b, a = iir.biquad_ap1st(frequency, fs, ai=coefficients)
+    elif order == 2:
+        if coefficients is None:
+            coefficients = [0.8832, 1.6278]
```

Comment:
```suggestion
            coefficients = [1.6278, 0.8832]
```
it does not match the documentation above.

Reply:
The coefficients in the docstring were swapped.